### PR TITLE
fix: Support logo feature for visual story themes

### PR DESCRIPTION
**Pull request by shraddha-kesari**
# Description

Added Support for logo feature on visual story when themes are added. 

Fixes # (issue-reference)
Added support if visual stories config is an array of objects
ref ticket: https://github.com/quintype/ace-planning/issues/659

Dependencies # (dependency-issue-reference)

Documentation # (link to the corresponding documentation changes)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test Case A
- [ ] Test Case B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 1]
```diff
@@ -18,16 +18,34 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+        customTemplatelogoAlignment = visualStoriesConfigWithThemes[0] && (typeof visualStoriesConfigWithThemes[0].logoAlignment === "function" ? visualStoriesConfigWithThemes[0].logoAlignment(config) : visualStoriesConfigWithThemes[0].logoAlignment);
+        customTemplatelogoUrl = visualStoriesConfigWithThemes[0] && (typeof visualStoriesConfigWithThemes[0].logoUrl === "function" ? visualStoriesConfigWithThemes[0].logoUrlThemeBased(config) : visualStoriesConfigWithThemes[0].logoUrl);
+    }
+  }
+
+  const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
+  const logoAlignment = typeof visualStoriesConfig.logoAlignment === "function" ? visualStoriesConfig.logoAlignment(config) : visualStoriesConfig.logoAlignment;
+  const logoUrl = typeof visualStoriesConfig.logoUrl === "function" ? visualStoriesConfig.logoUrl(config) : visualStoriesConfig.logoUrl;
+
+  const visualStoryConfig = { logoUrl, logoAlignment, customTemplatelogoAlignment , customTemplatelogoUrl };
```
Comment (ags1773) [inline src/atoms/visual-story/cover-page/cover-page.tsx:48]: don't pass customTemplatelogoAlignment to child component. The decision to use which logo url should be done here itself. Clean up this code a little, there's some repetition

```js
const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
const visualStoriesConfigWithThemes = get(config, ["opts", "featureConfig", "visualStories"]) || [];
```

**Comment by shraddha-kesari** [comment]
Check this feature on `fix-logo` branch on malibu advanced

**Review by ags1773 — APPROVED** [review]

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 2]
```diff
@@ -18,15 +18,29 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
-      ? visualStoriesConfig.logoAlignment(config)
-      : visualStoriesConfig.logoAlignment;
-  const logoUrl =
-    typeof visualStoriesConfig.logoUrl === "function"
-      ? visualStoriesConfig.logoUrl(config)
-      : visualStoriesConfig.logoUrl;
+  const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
+
+  let logoAlignment = visualStoriesConfig && (typeof visualStoriesConfig.logoAlignment === "function" ? visualStoriesConfig.logoAlignment(config) : visualStoriesConfig.logoAlignment) || null;
+  let logoUrl = visualStoriesConfig && (typeof visualStoriesConfig.logoUrl === "function" ? visualStoriesConfig.logoUrl(config) : visualStoriesConfig.logoUrl) || null;
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:24]: refactor to much readable code

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 3]
```diff
@@ -18,15 +18,29 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+  
+  if (Array.isArray(visualStoriesConfig)) {
+    const visualStoryTheme = get(story, ["metadata", "story-attributes", "visualstorytheme"]) || [];
+    const theme = visualStoryTheme[0];
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:28]: use `get`

File: src/atoms/publisher-logo-header/publisher-logo-header.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ export const PublisherLogoHeaderBase = ({ config, visualStoryConfig }: Publisher
   const publisherName = get(config, ["publisherConfig", "publisher-name"], "");
   const logo = get(config, ["ampConfig", "logo-url"], null);
   const logoAlignment = visualStoryConfig && visualStoryConfig.logoAlignment;
-  const logoUrl = (visualStoryConfig && visualStoryConfig.logoUrl) || logo;
+  const logoUrl = visualStoryConfig && visualStoryConfig.logoUrl  || logo;
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply src/atoms/publisher-logo-header/publisher-logo-header.tsx:11]: Okay sure.

[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply src/atoms/publisher-logo-header/publisher-logo-header.tsx:11]: Any particular reason, why `get`  method and not `&&`??

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 5]
```diff
@@ -18,15 +18,29 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+      case "theme-3":
+        logoAlignment = visualStoriesConfig[2] && (typeof visualStoriesConfig[2].logoAlignment === "function" ? visualStoriesConfig[2].logoAlignment(config) : visualStoriesConfig[2].logoAlignment);
+        logoUrl = visualStoriesConfig[2] && (typeof visualStoriesConfig[2].logoUrl === "function" ? visualStoriesConfig[2].logoUrl(config) : visualStoriesConfig[2].logoUrl);
+        break;
+      default:
+        logoAlignment = visualStoriesConfig[0] && (typeof visualStoriesConfig[0].logoAlignment === "function" ? visualStoriesConfig[0].logoAlignment(config) : visualStoriesConfig[0].logoAlignment);
+        logoUrl = visualStoriesConfig[0] && (typeof visualStoriesConfig[0].logoUrl === "function" ? visualStoriesConfig[0].logoUrl(config) : visualStoriesConfig[0].logoUrl);
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 3 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Changed

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 6]
```diff
@@ -18,15 +18,29 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
-  const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"], {});
-  const logoAlignment =
-    typeof visualStoriesConfig.logoAlignment === "function"
-      ? visualStoriesConfig.logoAlignment(config)
-      : visualStoriesConfig.logoAlignment;
-  const logoUrl =
-    typeof visualStoriesConfig.logoUrl === "function"
-      ? visualStoriesConfig.logoUrl(config)
-      : visualStoriesConfig.logoUrl;
+  const visualStoriesConfig = get(config, ["opts", "featureConfig", "visualStories"]) || {};
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 7]
```diff
@@ -19,14 +19,59 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+    const getLogoAlignmentTheme2 = isLogoAlignmentTheme2 ? visualStoriesConfig[2].logoAlignment(config) : visualStoriesConfig[2].logoAlignment;
+    const getLogoAlignmentDefaultTheme = isLogoAlignmentDefaultTheme ? visualStoriesConfig[0].logoAlignment(config) : visualStoriesConfig[0].logoAlignment;
+
+    // To support PB config
+    const isLogoUrlTheme1 = typeof visualStoriesConfig[1].logoUrl === "function";
+    const isLogoUrlTheme2 = typeof visualStoriesConfig[2].logoUrl === "function";
+    const isLogoUrlDefaultTheme = typeof visualStoriesConfig[0].logoUrl === "function";
+
+    const getLogoUrlTheme1 = isLogoUrlTheme1 ? visualStoriesConfig[1].logoUrl(config) : visualStoriesConfig[1].logoUrl;
+    const getLogoUrlTheme2 = isLogoUrlTheme2 ? visualStoriesConfig[2].logoUrl(config) : visualStoriesConfig[2].logoUrl;
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:52]: DRY seems to be missing

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 8]
```diff
@@ -19,14 +19,59 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+       
+      case "theme-3": {
+        logoAlignment = visualStoriesConfig[2] && getLogoAlignmentTheme2;
+        logoUrl = visualStoriesConfig[2] && getLogoUrlTheme2;
+        break;
+      }
+      default: {
+        logoAlignment = visualStoriesConfig[0] && getLogoAlignmentDefaultTheme;
+        logoUrl = visualStoriesConfig[0] && getLogoUrlDefaultTheme;
+      }
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:71]: DRY

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 1 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 8 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 7 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 9]
```diff
@@ -19,16 +19,53 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+        logoAlignment = visualStoriesConfig[2] && getLogoAlignmentTheme;
+        logoUrl = visualStoriesConfig[2] && getLogoUrlTheme;
+        break;
+      }
+      default: {
+        logoAlignment = visualStoriesConfig[0] && getLogoAlignmentTheme;
+        logoUrl = visualStoriesConfig[0] && getLogoUrlTheme;
+      }
+    }
+  }
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:65]: Can this be written better? Seems to repeat

File: src/atoms/visual-story/cover-page/cover-page.tsx [hunk 10]
```diff
@@ -19,16 +19,53 @@ export const CoverPageBase = ({ story, config }: CoverPageProps) => {
...
+  let logoUrl = getLogoUrl;
+
+  if (Array.isArray(visualStoriesConfig)) {
+    const visualStoryTheme = get(story, ["metadata", "story-attributes", "visualstorytheme"], []);
+    const theme = get(visualStoryTheme, [0]);
+    
+    // if theme 2, it will pick up value from 2nd array
+    // if theme 3, it will pick up value from 3rd array
+    // else, it will pick up from 1st array
+    const isTheme = theme === "theme-2" ? 1 : theme === "theme-3" ? 2 : 0;
```
Comment (Jeevan-Kishore) [inline src/atoms/visual-story/cover-page/cover-page.tsx:40]: Do not batch ternary conditions

File: package.json [hunk 11]
```diff
@@ -93,7 +93,9 @@
   },
   "dependencies": {
     "@rvgpl/get-youtube-id": "^1.0.0",
+    "airbnb-browser-shims": "^3.3.0",
```
Comment (Jeevan-Kishore) [inline package.json:96]: Why are these required?

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 9 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Removed the switch case

File: src/atoms/visual-story/cover-page/cover-page.tsx (hunk 10 above)
Comment (shraddha-kesari) [reply src/atoms/visual-story/cover-page/cover-page.tsx]: Done

File: package.json (hunk 11 above)
Comment (shraddha-kesari) [reply package.json]: Not needed now. Uninstalled.

**Review by Jeevan-Kishore — APPROVED** [review]
